### PR TITLE
Update dependencies; update README.md

### PR DESCRIPTION
**Pull request by bgola-signalfx**
Key changes:
 - updated SignalFx client and SignalFx tracing libs to the latest version
 - updated README.md with sample code demonstrating how to add extra tags to spans sent by the wrapper
 - updated README.md with recommended approach to metrics and events delivery

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,13 +190,12 @@ To learn more, see:
...
-         .then(() => callback(null, 'Done'));
+      signalFxLambda.helper.sendCustomEvent('Custom', {functionName: context.functionName}, {description: 'Custom event'});
       
       // to transform & forward CloudWatch event:
-      signalFxLambda.helper.sendCloudWatchEvent(event)
-         .then(() => callback(null, 'Done'))
-      
+      signalFxLambda.helper.sendCloudWatchEvent(event);
+         
+      callback(null, 'Done');
```
Comment (owais) [inline README.md:198]: Why was `.then` removed? Wouldn't this possibly result in the event being lost if the function stopped before the event was exported successfully?

File: README.md [hunk 2]
```diff
@@ -209,10 +208,10 @@ To learn more, see:
     exports.handler = signalFxLambda.asyncWrapper(async (event, context) => {
       ...
       // to send custom event:
-      await signalFxLambda.helper.sendCustomEvent('Custom', {functionName: context.functionName}, {description: 'Custom event'});
+      signalFxLambda.helper.sendCustomEvent('Custom', {functionName: context.functionName}, {description: 'Custom event'});
             
       // to transform & forward CloudWatch event:
-      await signalFxLambda.helper.sendCloudWatchEvent(event);
+      signalFxLambda.helper.sendCloudWatchEvent(event);
```
Comment (owais) [inline README.md:214]: same comment as above. If we don't wait for the function to finish exporting, there is no guarantee the data will actually be reported as function may exit and execution may be stopped before the metrics lib has a chance to complete the export.

File: README.md [hunk 3]
```diff
@@ -265,6 +264,36 @@ The tracing wrapper creates a span for the wrapper handler. This span contains t
...
+});
+```
+
+### Asynchronous Metrics and Events Delivery
+
+Although the following `signalFxLambda.helper.send*` methods all return promises that resolve when send operation completes it is recommended to ignore those promises as the wrapper is going to wait for their completion internally.
+
+This way multiple send operations may run concurrently without blocking your lambda handler code.
+
+You can still await any of those promises if your use case forces you to do so.
```
Comment (owais) [inline README.md:288]: OK. This answers my questions above :)

**Review by owais — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (bgola-signalfx) [reply README.md:288]: Great, thanks! :)

**Review by piotr-signalfx — APPROVED** [review]

**Pull request merged**